core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1231,7 +1234,8 @@ private[spark] class DAGScheduler(
       logInfo(s"Submitting ${tasks.size} missing tasks from $stage (${stage.rdd}) (first 15 " +
         s"tasks are for partitions ${tasks.take(15).map(_.partitionId)})")
       taskScheduler.submitTasks(new TaskSet(
-        tasks.toArray, stage.id, stage.latestInfo.attemptNumber, jobId, properties))
+        tasks.toArray, stage.id, stage.latestInfo.attemptNumber, jobId, properties,
```

Comment:
What about making `Stage#nextAttemptId` readable and using it instead of `stage.latestInfo.attemptNumber`?

Reply:
Maybe `stage.latestInfo.attemptNumber` is better for readability, nextAttemptId = latestInfo.attemptNumber + 1.